### ISSUE TITLE
Crash after leaving sub through airlock

### ISSUE THREAD

**Issue by diego-caripan** [Regalis11/Barotrauma#9382]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Loaded a campaign i was playing, at Great Sea biome. Had a Hunting Ground mission. Boss was a Endworm. After i killed it, and repaired my sub (Remora), tried to mine some minerals from nearby abyss islands. However, every time i try to leave the sub through the airlock (only after i kill the boss), my game crashes after a few seconds out of the sub. Not sure if only happens in this biome, or on every one where there's a Hunting Grounds mission. Tried this 3 times, and 3 times it crashed, so the logs are similar. Uploaded the lastest one.

It appears it has to do with the pump at the airlock.

(Seems someone has a similar problem, but happened with a different action. [Here](https://github.com/Regalis11/Barotrauma/issues/9371) )

### Reproduction steps

1. Start campaign at Great Sea, Hunting Grounds with Endworm.
2. Kill Endworm.
3. Try to leave the sub through airlock.
4. Game crashes.

### Bug prevalence

Happens every time I play

### Version

0.18.11.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Barotrauma Client crash report (generated on 16-06-2022 6:57:18)


Barotrauma seems to have crashed. Sorry for the inconvenience! 


126A1EFE280FCB5E98C29D91D4C57FA1


Game version 0.18.11.0 (ReleaseWindows, branch release, revision 75c2f92d56)
Graphics mode: 1920x1080 (Fullscreen)
VSync ON
Language: English
Selected content packages: Vanilla
Level seed: Kennet LineaKimju
Loaded submarine: Remora (89139927B8E6FB9CECC93581196859C2)
Selected screen: Barotrauma.GameScreen
SteamManager initialized


System info:
    Operating system: Microsoft Windows NT 10.0.19044.0 64 bit
    GPU name: NVIDIA GeForce RTX 3060
    Display mode: {Width:1920 Height:1080 Format:Color AspectRatio:1,7777778}
    GPU status: Normal


Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Boolean Filter(Barotrauma.Items.Components.Pump)
Stack trace: 
   at Barotrauma.AIObjectivePumpWater.Filter(Pump pump) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectivePumpWater.cs:line 40
   at Barotrauma.AIObjectiveLoop`1.FindTargets() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectiveLoop.cs:line 187
   at Barotrauma.AIObjectiveLoop`1.UpdateTargets() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectiveLoop.cs:line 173
   at Barotrauma.AIObjectiveLoop`1.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectiveLoop.cs:line 77
   at Barotrauma.AIObjectiveManager.UpdateObjectives(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectiveManager.cs:line 271
   at Barotrauma.HumanAIController.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\HumanAIController.cs:line 396
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 2750
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 190
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 874
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 656
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 92
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Microsoft.Xna.Framework.Game.Run() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 367
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58


Last debug messages:
[16-06-2022 6:23:44] Could not find a spawn point for a SpawnAction (spawn location: BeaconStation (tag: beacon_bandit), skipping.
[16-06-2022 6:23:37] Generating an outpost for the end of the level... (Location: Kimju Outpost, level type: LocationConnection)
[16-06-2022 6:23:37] Generating an outpost for the start of the level... (Location: Kennet Linea, level type: LocationConnection)
[16-06-2022 6:23:34] Level identifier: greatseabasic
[16-06-2022 6:23:24] Transitioning to Kennet LineaKimju (current location: Kennet Linea, selected location: Kimju Outpost, leaving sub: Remora, at start: True, at end: False, transition type: LeaveLocation)
[16-06-2022 6:04:48] Alc.CaptureOpenDevice attempt 2 failed: error code 40964
[16-06-2022 6:04:48] alcCaptureOpenDevice failed: exception thrown (Device init failed: 0x80004005)
[16-06-2022 6:04:48] WASAPI capture open failed: HRESULT 0x80004005 (1)
[16-06-2022 6:04:48] Alc.CaptureOpenDevice attempt 1 failed: error code 40964
[16-06-2022 6:04:48] alcCaptureOpenDevice failed: exception thrown (Device init failed: 0x80004005)
[16-06-2022 6:04:48] WASAPI capture open failed: HRESULT 0x80004005 (1)
[16-06-2022 6:03:10] Alc.CaptureOpenDevice attempt 2 failed: error code 40964
[16-06-2022 6:03:10] alcCaptureOpenDevice failed: exception thrown (Device init failed: 0x80004005)
[16-06-2022 6:03:10] WASAPI capture open failed: HRESULT 0x80004005 (1)
[16-06-2022 6:03:10] Alc.CaptureOpenDevice attempt 1 failed: error code 40964
[16-06-2022 6:03:10] alcCaptureOpenDevice failed: exception thrown (Device init failed: 0x80004005)
[16-06-2022 6:03:10] WASAPI capture open failed: HRESULT 0x80004005 (1)
[16-06-2022 6:02:26] Unlocked mission "Terminate a Thresher Swarm" in the location "Kennet Linea".
[16-06-2022 6:02:21] Unlocked mission "Scan ruin" in the location "Kennet Linea".
[16-06-2022 6:02:18] Unlocked mission "Killing Hammerheads" in the location "Kennet Linea".
[16-06-2022 6:02:16] Unlocked mission "Pirate Vessel" in the location "Kennet Linea".
[16-06-2022 6:01:30] Generating an outpost for the start of the level... (Location: Kennet Linea, level type: Outpost)
[16-06-2022 6:01:30] Level identifier: outpostlevel
[16-06-2022 6:00:58] Attempting to open ALC device "OpenAL Soft on Altavoces (Realtek(R) Audio)"
[16-06-2022 6:00:57] Logged in as Dummy (SteamID STEAM_1:0:39195529)
```

**Comment by Regalis11** [comment]
Duplicate of https://github.com/Regalis11/Barotrauma/issues/9371
